tuist/tuist
# Script variable path error fixed

Resolves https://github.com/tuist/tuist/issues/1870 (this issue holds our team back from using **focus** feature.)

### Short description 📝

Since the script paths are defined as `AbsolutePath`, Xcode's environment variables can not be resolved properly, tuist just prepends the target root path as a prefix, which breaks tuist cache and tuist test subcommands.

This PR treats those paths as strings as they may have some variables which can only be resolved by Xcode (i.e `$(SRCROOT)`).

#### More Info:

When any of the following fields are populated in a `TargetScript`

-     public let inputPaths: [AbsolutePath]
-     public let inputFileListPaths: [AbsolutePath]
-     public let outputPaths: [AbsolutePath]
-     public let outputFileListPaths: [AbsolutePath]

like in the below

```swift
 .pre(
      path: .relativeToRoot("script.sh"),
      arguments: [],
      name: "My Script",
      inputFileListPaths: ["$(SRCROOT)/MyTarget.xcfilelist"],
      outputPaths: ["$(DERIVED_FILE_DIR)/script_ran.txt"]
    )
```

`tuist generate` can generate the project and things work as expected until we try to run `tuist test` or `tuist cache warm`. Then we get the following error

```swift
File not found at /Users/eekin/Desktop/m1/MyApp/MyTarget/$(SRCROOT)/MyTarget.xcfilelist
File not found at /Users/eekin/Desktop/m1/MyApp/MyTarget/$(DERIVED_FILE_DIR)/script_ran.txt.txt
```

### How to test the changes locally 🧐

```bash
$ mkdir baklava
$ cd baklava
$ tuist init --platform ios
$ tuist edit
```

add a script phase with output path, including an Xcode build environment variables such as `DERIVED_FILE_DIR`
(see more env variables by running `xcodebuild -showBuildSettings`)

```swift
  sources: ["Targets/\(name)/Sources/**"], resources: [], scripts: [
                          .pre(
                            script: "ls -la > ${SCRIPT_OUTPUT_FILE_0}",
                            name: "",
                            outputPaths: ["$(DERIVED_DILE_DIR)/script_ran.txt"]
                          )
                         ],
```

```bash
$ tuist test
```

You'll see the logs as

```bash
Generating project for testing
File not found at /Users/eekin/Desktop/baklava/$(DERIVED_DILE_DIR)/script_ran.txt
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

### Checklist ✅
